src/common/heartbeat2/src/client.cpp

```diff
@@ -0,0 +1,137 @@
+    }
+
+    void handshake () {
+        // checking if service is available or waiting for service to be availble
+        RCLCPP_INFO(this->get_logger(), "Waiting for drivestation.");
+        while (!handshakeClient->wait_for_service(std::chrono::milliseconds(10))) {
+            if (!rclcpp::ok()) {
+                RCLCPP_ERROR(this->get_logger(), "Interrupted while waiting");
+                exit(1);
+            }
+        }
+        RCLCPP_INFO(this->get_logger(), "Found drivestation");
+
+        // Make handshake request and send
+        auto request = std::make_shared<csm_common_interfaces::srv::Heartbeat::Request>();
+        request->milliseconds = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+        _handshakeFuture = std::move(handshakeClient->async_send_request(request).future.share());
+
+        // wait for response   
+        handshakeTimer->reset();
+    }
+
+private:
+    void handshake_callback() {
+        // check for handshake response
+        if (_handshakeFuture.wait_for(std::chrono::milliseconds(100)) == std::future_status::ready) {
+            auto response = _handshakeFuture.get();
+            _lastResponseTime = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+            RCLCPP_INFO(this->get_logger(), "Connected to drivestation");
+            handshakeTimer->cancel();
+            pingTimer->reset();
+        } 
+        if (!rclcpp::ok()) {
+            RCLCPP_ERROR(this->get_logger(), "Interrupted while waiting for response");
+            exit(2);
+        }
+    }
+
+
+    void ping_callback() {
+        unsigned int currentMillisec = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+        // send a request every 300ms
+        if (heartbeatClient->wait_for_service(std::chrono::milliseconds(1))) {
+            auto request = std::make_shared<csm_common_interfaces::srv::Heartbeat::Request>();
+            request->milliseconds = currentMillisec;
+            _futures->push_back(std::move(heartbeatClient->async_send_request(request).future.share()));
+
+            // debug
+            // RCLCPP_INFO(this->get_logger(), "Sent ping at %u", currentMillisec);
+        }
+        
+        // check for responses
+        int futuresSize = _futures->size();
+        int increment = 0;
+        std::shared_ptr<csm_common_interfaces::srv::Heartbeat::Response> response;
+        for (int i = 0; i < futuresSize; i++) {
+            if (_futures->at(increment).wait_for(std::chrono::milliseconds(1)) == std::future_status::ready) {
+                // if future is ready, check if the time sent is latest, and update latest time
+                response = _futures->at(increment).get();
+                _futures->erase(_futures->begin() + increment);
+                _lastResponseTime = currentMillisec;
```

Comment:
If we aren't using the future's response, why do we store it as a uint64? Why not just an Empty or bool?